backend/test/test_utils.go

```diff
@@ -188,3 +216,83 @@ func GetDefaultPipelineRunnerServiceAccount(isKubeflowMode bool) string {
 		return "pipeline-runner"
 	}
 }
+
+// Checks if the resRefs contain the targets. Ignores resource's Name. Ignores duplicate or nil entries in targets.
+func VerifyExperimentResourceReferences(resRefs []*experiment_model.APIResourceReference, targets []*experiment_model.APIResourceReference) bool {
+	matches := 0
+	for _, target := range targets {
+		for _, resRef := range resRefs {
+			if target.Key == nil {
+				matches++
```

Comment:
This is to cover the case when:
```golang
targetsNIL := []*experiment_model.APIResourceReference{
    &experiment_model.APIResourceReference{},
}

targetsEMPTY := []*experiment_model.APIResourceReference{
}
```
Both cases are equivalent as they have empty references, but `len(targetsNIL)==1` while `len(targetsEMPTY)==0`. The logic makes sure that the function returns `true`. This may not happen in the tests as of today though. I rearranged it a little for better readability.

Reply:
I see your logic now. What you have is fine here, it would likely be challenged in a coding interview though :)
For the described purpose, I'd typically expect some code like below: 
```golang
func checkExists(target *APIResourceReference, resRefs []*APIResourceReference) bool {
    for _, resRef := range resRefs {
        if target.Key.ID == resRef.Key.ID && ... {
            return true
        }
    }
    return false
}

func VerifyExperimentResourceReferences(...) bool {
    for _, target := range targets {
        if target.key == nil {
            continue
        }
        if !checkExists(target, resRefs) {
            return false
        }
    }
    return true
}
```

Beside, for my question: "what are some examples of resource references with a key being nil?"
I think I could image when users don't pass any resource reference, we would get
```golang
targetsEMPTY := []*experiment_model.APIResourceReference{
}
```
But I'm not sure if any user scenario would result
```golang
targetsNIL := []*experiment_model.APIResourceReference{
    &experiment_model.APIResourceReference{},
}
```